lib/fastlane/actions/ipa.rb

```diff
@@ -0,0 +1,112 @@
+    # -d, --destination DESTINATION Destination. Defaults to current directory 
+    # -m, --embed PROVISION Sign .ipa file with .mobileprovision 
+    # -i, --identity IDENTITY Identity to be used along with --embed 
+    # --sdk SDK            use SDK as the name or path of the base SDK when building the project 
+    # --ipa IPA            specify the name of the .ipa file to generate (including file extension) 
+
+    ARGS_MAP = {
+      workspace: '-w',
+      project: '-p',
+      configuration: '-c',
+      scheme: '-s',
+      clean: '--clean',
+      archive: '--archive',
+      destination: '-d',
+      embed: '-m',
+      identity: '-i',
+      sdk: '--sdk',
+      ipa: '--ipa'
+    }
+
+    class IpaAction
+      def self.run(params)
+        
+        # The args we will build with
+        build_args = nil
+
+        # The output directory of the IPA and dSYM
+        absolute_dest_directory = nil
+
+        # Allows for a whole variety of configurations
+        if params.first.is_a? Hash
+
+          # Used to get the final path of the IPA and dSYM
+          if dest = params.first[:destination]
+            absolute_dest_directory = Dir.glob(dest).map(&File.method(:realpath)).first
+          end
+
+          # Maps nice developer build parameters to Shenzhen args
+          build_args = params_to_build_args(params.first)
+
+        else
+          build_args = params
+        end
+
+        # If no dest directory given, default to current directory
+        absolute_dest_directory ||= Dir.pwd
+
+        if Helper.is_test?
+          Actions.lane_context[SharedValues::IPA_OUTPUT_PATH] = File.join(absolute_dest_directory, "test.ipa")
+          Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] = File.join(absolute_dest_directory, "test.app.dSTM.zip")
+          return build_args 
+        end
+
+        # Joins args into space delimited string
+        build_args = build_args.join(' ')
+
+        Actions.sh "ipa build #{build_args}"
+        
+        # Finds absolute path of IPA and dSYM
+        absolute_ipa_path = find_ipa_file(absolute_dest_directory)
+        absolute_dsym_path = find_dsym_file(absolute_dest_directory)
+
+        # Sets shared values to use after this action is performed
+        Actions.lane_context[SharedValues::IPA_OUTPUT_PATH] = absolute_ipa_path
+        Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] = absolute_dsym_path
+
+
+      end
+
+      def self.params_to_build_args(params)
+        # Remove nil value params unless :clean or :archive
+        params = params.delete_if { |k, v| (k != :clean && k != :archive ) && v.nil? }
+
+        # Maps nice developer param names to Shenzhen's `ipa build` arguments
+        params.collect do |k,v|
+          v ||= ''
+          if args = ARGS_MAP[k]
+           "#{ARGS_MAP[k]} #{v}".strip
+          end
+        end.compact
+      end
+
+      def self.find_ipa_file(dir)
+        # Finds last modified .ipa in the destination directory
```

Comment:
Maybe we should make use of the change with my recent pull request: https://github.com/nomad/shenzhen/pull/196

That means, the absolute path to the IPA is shown as the last line of `shenzhen`. What do you think?